linters/trivy/trivy.test.ts

```diff
@@ -1,16 +1,65 @@
 import * as fs from "fs";
 import path from "path";
-import { fuzzyLinterCheckTest } from "tests";
+import { customLinterCheckTest, fuzzyLinterCheckTest, TestCallback } from "tests";
+import { TrunkLintDriver } from "tests/driver";
 import { FileIssue } from "tests/types";
+import { TEST_DATA } from "tests/utils";
 import { createFuzzyMatcher } from "tests/utils/landing_state";
 
-// trunk-ignore(eslint/@typescript-eslint/no-unsafe-assignment)
-const expectedFileIssues = JSON.parse(
-  fs.readFileSync(path.resolve(__dirname, "expected_issues.json")).toString()
-);
+const callbackGenerator =
+  (command: string, otherPreCheck?: (driver: TrunkLintDriver) => void): TestCallback =>
+  (driver) => {
+    const trunkYamlPath = ".trunk/trunk.yaml";
+    const currentContents = driver.readFile(trunkYamlPath);
+    const trivyRegex = /- trivy@(.+)\n/;
+    const newContents = currentContents.replace(
+      trivyRegex,
+      `- trivy@$1:\n        commands: [${command}]\n`,
+    );
+    driver.writeFile(trunkYamlPath, newContents);
+    if (otherPreCheck) {
+      otherPreCheck(driver);
+    }
+  };
+
+const configExpectedFileIssues = JSON.parse(
+  fs.readFileSync(path.resolve(__dirname, "config_expected_issues.json")).toString(),
+) as FileIssue[];
```

Comment:
nit: I would actually prefer this as a magic number rather than parsing here, just in case if we mess it up accidentally, we'll be asserting something like a 0 issue match, which will always be true